src/lib/dialog/dialog-container.ts

```diff
@@ -118,11 +124,20 @@ export class MdDialogContainer extends BasePortalHost implements OnDestroy {
     // If were to attempt to focus immediately, then the content of the dialog would not yet be
     // ready in instances where change detection has to run first. To deal with this, we simply
     // wait for the microtask queue to be empty.
-    this._elementFocusedBeforeDialogWasOpened = document.activeElement as HTMLElement;
     this._focusTrap.focusFirstTabbableElementWhenReady();
   }
 
   /**
+   * Saves a reference to the element that was focused before the dialog was opened.
+   * @private
+   */
+  private _savePreviouslyFocusedElement() {
+    if (this._document) {
+      this._elementFocusedBeforeDialogWasOpened = this._document.activeElement as HTMLElement;
+    }
+  }
```

Comment:
It was tricky to write a test that fails when we expect it to since we trigger the change detection and animations manually.